### PR TITLE
New chip styling and white chip variant

### PR DESCRIPTION
**Pull request by mariush2**
Also refactored how the prop types for the different chips are defined. Now you can't set both onDelete and onClick (will throw TS error)

---

Userstories:
- [AB#528554](https://dev.azure.com/Equinor/1f7078da-0ba3-4461-a220-c3e39c5c1f09/_workitems/edit/528554)
- [AB#529794](https://dev.azure.com/Equinor/1f7078da-0ba3-4461-a220-c3e39c5c1f09/_workitems/edit/529794)

Tasks:
- [AB#533017](https://dev.azure.com/Equinor/1f7078da-0ba3-4461-a220-c3e39c5c1f09/_workitems/edit/533017)
- [AB#533018](https://dev.azure.com/Equinor/1f7078da-0ba3-4461-a220-c3e39c5c1f09/_workitems/edit/533018)
- [AB#533019](https://dev.azure.com/Equinor/1f7078da-0ba3-4461-a220-c3e39c5c1f09/_workitems/edit/533019)

### CODE REVIEW

File: src/molecules/Chip/InteractiveChip.tsx [hunk 1]
```diff
@@ -3,51 +3,46 @@ import { forwardRef, HTMLAttributes, KeyboardEvent } from 'react';
...
   const handleKeyPress = (event: KeyboardEvent<HTMLButtonElement>) => {
     const { key } = event;
     if (key === 'Enter') {
-      if (deletable) {
-        handleDelete?.(event); // using non-null assertion operator since we know deletable is true
-      } else if (clickable) {
-        handleClick?.(event); // using non-null assertion operator since we know clickable is true
+      if ('onClick' in props) {
+        relevantOnAction(event);
       }
```
Comment (aslakihle) [inline src/molecules/Chip/InteractiveChip.tsx:35]: If you focus a deletable chip, and press enter, i dont think onDelete will be called, but i think it should be

**Review by aslakihle — CHANGES_REQUESTED** [review]

File: src/molecules/Chip/Chip.stories.tsx [hunk 2]
```diff
@@ -48,6 +55,7 @@ const meta: Meta<typeof Chip> = {
         type: 'radio',
         options: [
           'default',
+          'white',
           'active',
           'warning',
           'warning-active',
```
[the remark this answers is not in the document]
Comment (mariush2) [reply src/molecules/Chip/Chip.stories.tsx:61]: Pushed new commit

File: src/molecules/Chip/Chip.styles.tsx [hunk 3]
```diff
@@ -160,10 +177,11 @@ const commonChipStyle = css`
...
 
 export const InteractiveChipStyle = styled.button<InteractiveChipProps>`
+  cursor: pointer;
   ${commonChipStyle}
-
-  ${({ variant }) => {
-    const colorScheme = getColorSchemeBy(variant);
+  ${({ variant, selected }) => {
+    // White variant has the same active styling as default
+    const colorScheme = getColorSchemeBy(variant, selected);
```
[the remark this answers is not in the document]
Comment (mariush2) [reply src/molecules/Chip/Chip.styles.tsx]: New commit

File: src/molecules/Chip/InteractiveChip.tsx (hunk 1 above)
Comment (mariush2) [reply src/molecules/Chip/InteractiveChip.tsx]: New commit

**Review by aslakihle — DISMISSED** [review]

**Review by aslakihle — APPROVED** [review]

**Pull request merged**